src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt

```diff
@@ -91,55 +88,57 @@ data class RsQualifiedName private constructor(
 
     /** Tries to find rust qualified name element related to this qualified name */
     fun findPsiElement(psiManager: PsiManager, context: RsElement): RsQualifiedNamedElement? {
-        val item = childItems.lastOrNull() ?: parentItem
-
-        // If it's link to crate, try to find the corresponding `lib.rs` file
-        if (item.type == CRATE) {
-            return findCrateRoot(item, psiManager, context)
+        /** `this` is absolute path even for relative links, conversion is done in [documentationAsHtml] */
+        val crateRoot = findCrateRoot(crateName, psiManager, context) ?: return null
+        return when (val itemType = itemType) {
+            CRATE -> crateRoot
+            FIELD, STRUCTFIELD -> resolveAsField(crateRoot)
+            else -> resolveAsPsiElement(itemType, crateRoot)
         }
+    }
 
-        // Otherwise, split search into two steps:
-        val project = context.project
-        // First: look for `RsQualifiedNamedElement` with the same name as expected,
-        // generate sequence of all possible reexports of this element and check
-        // if any variant has the same `RsQualifiedName`
-        var result = lookupInIndex(project, RsNamedElementIndex.KEY, { it }) { element ->
-            if (element !is RsQualifiedNamedElement) return@lookupInIndex null
-            val candidate = if (element is RsModDeclItem && item.type == MOD) {
-                element.reference.resolve() as? RsMod ?: return@lookupInIndex null
-            } else {
-                element
-            }
-            QualifiedNamedItem.ExplicitItem(candidate)
+    private fun resolveAsPsiElement(itemType: ItemType, crateRoot: RsFile): RsQualifiedNamedElement? {
```

Comment:
Personally, I find this name is confusing without additional explanation because the name is almost the same as `findPsiElement`

Maybe rename to somthing like `resolveToPathBasedElement` and add short explanation that it can be used for any element that can be represented as some valid Rust path unlike field